keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -260,64 +256,53 @@ private void closeAwsSecretsManagers() {
     awsSecretsManagerInvalidCredentials.close();
   }
 
-  private void closeClients() {
-    closeAwsSecretsManagers();
-    closeTestSecretsManager();
-  }
-
-  private String createSecret(final Tag tag, String secretValue)
-      throws ExecutionException, InterruptedException, TimeoutException {
-    final String testSecretName = SECRET_NAME_PREFIX + UUID.randomUUID();
-
-    final CreateSecretRequest secretRequest =
-        CreateSecretRequest.builder()
-            .name(testSecretName)
-            .secretString(secretValue)
-            .tags(tag)
-            .build();
-
-    testSecretsManagerClient.createSecret(secretRequest).get(30, TimeUnit.SECONDS);
-    waitUntilSecretAvailable(testSecretName);
-    return testSecretName;
+  private void createTestSecrets() {
+    createOrUpdateSecret(SECRET_NAME1_KEY1_VALA, "tagKey1", "tagValA", SECRET_VALUE1);
+    createOrUpdateSecret(SECRET_NAME2_KEY1_VALB, "tagKey1", "tagValB", SECRET_VALUE2);
+    createOrUpdateSecret(SECRET_NAME3_KEY2_VALC, "tagKey2", "tagValC", SECRET_VALUE3);
+    createOrUpdateSecret(SECRET_NAME4_KEY2_VALB, "tagKey2", "tagValB", SECRET_VALUE4);
   }
 
-  private String createTestSecret(
-      final String tagKey, final String tagVal, final String secretValue) throws Exception {
+  private void createOrUpdateSecret(
+      final String testSecretName,
+      final String tagKey,
+      final String tagVal,
+      final String secretValue) {
     final Tag testSecretTag = Tag.builder().key(tagKey).value(tagVal).build();
     try {
-      return createSecret(testSecretTag, secretValue);
-    } catch (Exception e) {
-      throw new Exception(e.getMessage());
+      final GetSecretValueResponse getSecretValueResponse =
+          testSecretsManagerClient.getSecretValue(
+              GetSecretValueRequest.builder().secretId(testSecretName).build());
+      final DescribeSecretResponse describeSecretResponse =
+          testSecretsManagerClient.describeSecret(
+              DescribeSecretRequest.builder().secretId(testSecretName).build());
+      final boolean hasDifferentSecretValue =
+          !getSecretValueResponse.secretString().equals(secretValue);
+      final boolean hasDifferentSecretTag =
+          describeSecretResponse.hasTags()
+              && !describeSecretResponse.tags().get(0).equals(testSecretTag);
+      if (hasDifferentSecretValue || hasDifferentSecretTag) {
```

Comment:
Think it might need to be: `hasDifferentSecretValue || hasNoTags || hasDifferentSecretTag`
to cover the case where a tag is removed, e.g. manually.